src/rules/extensions.js

```diff
@@ -46,9 +46,10 @@ module.exports = {
       typeof configuration === 'object' ? configuration : context.options[1]
     )
 
-    function isUseOfExtensionRequired(extension) {
+    function isUseOfExtensionRequired(extension, isModule) {
       if (!has(modifiers, extension)) { modifiers[extension] = defaultConfig }
-      return modifiers[extension] === 'always'
+      const modifier = modifiers[extension]
+      return (!isModule && modifier === 'ignorePackages') || modifier === 'always'
```

Comment:
I'd really prefer to avoid complex regexes if we can, but your matches look good.